src/labeler.ts

```diff
@@ -214,19 +261,31 @@ function checkAll(changedFiles: string[], globs: string[]): boolean {
 }
 
 function checkMatch(changedFiles: string[], matchConfig: MatchConfig): boolean {
-  if (matchConfig.all !== undefined) {
-    if (!checkAll(changedFiles, matchConfig.all)) {
-      return false;
+  if (matchConfig.changedFiles?.all !== undefined) {
+    if (checkAll(changedFiles, matchConfig.changedFiles.all)) {
+      return true;
     }
   }
 
-  if (matchConfig.any !== undefined) {
-    if (!checkAny(changedFiles, matchConfig.any)) {
-      return false;
+  if (matchConfig.changedFiles?.any !== undefined) {
+    if (checkAny(changedFiles, matchConfig.changedFiles.any)) {
+      return true;
```

Comment:
This change introduces a regression. There is an example in the README file:
```yml
# Add 'frontend` label to any change to *.js files as long as the `main.js` hasn't changed
frontend:
- any: ['src/**/*.js']
  all: ['!src/main.js']
```

Now, if `all` passes, `any` will not be checked.